src/hooks/use-snapshots.tsx

```diff
@@ -93,6 +100,8 @@ export const SnapshotProvider: React.FC< { children: ReactNode } > = ( { childre
 	const [ snapshotQuota, setSnapshotQuota ] = useState( LIMIT_OF_ZIP_SITES_PER_USER );
 	const [ isLoadingSnapshotUsage, setIsLoadingSnapshotUsage ] = useState( false );
 	const [ snapshotCreationBlocked, setSnapshotCreationBlocked ] = useState( false );
+	const [ isSnapshotProcessing, setIsSnapshotProcessing ] = useState( false );
+	const progress = useProgress( { isProcessing: isSnapshotProcessing } );
```

Comment:
What if we move the state inside the useProgress?
It could expose `startProgress` , `stopProgress` and `isActive`. Or maybe an object progress that contains all of that { start, stop, isActive, number }.

That way we could return. a single variable:
```
-			isSnapshotProcessing,
- 			setIsSnapshotProcessing,
			processingProgress, // this variable will contain start, stop and isActive
```

Reply:
I have made the refactor in https://github.com/Automattic/studio/pull/960/commits/8aa4e7284bf02ccb1a174e036c29e096170c3030, but it made things a bit more complicated with the progress for site deletion.

For now I am starting / stopping the progress there with `useEffect` as otherwise it looks like we would need to lift the deletion progress to the context as well.

But at the same time, the deletion is so quick that it may not be worth keeping track of the progress of each site that is being deleted separately (for the matter of switching tabs).